### DISCUSSION TITLE
Problems wrapping a Link around an SVG icon

### DISCUSSION THREAD

**Discussion by gmg** [diegomura/react-pdf#2451] [Q&A]
I am working on a linked digital notebook PDF for Kindle Scribe. There will be a sidebar with shortcuts to the different sections, including home and forward/back buttons. The section links (Link around Text element containing page numbers) are working as expected, on PC and on device. The other buttons are not working on the device.

I have created these buttons using FontAwesome icons, rendered as SVG using a variation of this solution: https://github.com/diegomura/react-pdf/discussions/1334

The navigation buttons are created by wrapping a Link and a View around the icon, like so:
```
<Link src={props.src} style={styles.buttonLink}>
    <View style={styles.button}>
        <FontAwesomeIcon
            faIcon={props.icon}
            style={{
                color: props.color,
                width: props.iconSize,
                height: props.iconSize
            }}
        />
    </View>
</Link>
```

I have tried replacing the icon with a standard block of text, and it works fine. My text buttons look like this:
```
<Link src={props.src} style={styles.buttonLink}>
    <View style={styles.button}>
        <Text style={styles.buttonText}>{props.label}</Text>
    </View>
</Link>
```
The documentation for Link says "Link’s can be nested inside a Text component, or being inside any other valid primitive" - Is SVG not a valid primitive for Links? What about the View wrapping around the SVG? Is there anything I can do to make this work, or do I need to rethink it?

To recap - text links work on PC and on Kindle Scribe. Icon links only work on PC. Icons are drawn using SVG, inside a View.
